src/llvm/type.cr

```diff
@@ -126,7 +126,12 @@ struct LLVM::Type
   end
 
   def const_int(value) : Value
-    Value.new LibLLVM.const_int(self, value, 0)
+    if !value.is_a?(Int128) && !value.is_a?(UInt128) && int_width != 128
+      Value.new LibLLVM.const_int(self, value, 0)
+    else
+      encoded_value = UInt64[value & UInt64::MAX, (value >> 64) & UInt64::MAX]
```

Comment:
But this allocates memory. Why not use static_array?

Reply:
AFAIK the reference is kept in llvm side. I couldn't find where the pointer was used to duplicate the data so I assume a reference was used. I could be wrong the documentation didn't said much.

Reply:
If it's kept in LLVM side then we must keep a reference on our side, otherwise the GC will collect it.

Reply:
It's copied: https://github.com/llvm-mirror/llvm/blob/master/lib/Support/APInt.cpp#L103

This should go on the stack.